crates/bitcoin-da/src/verifier.rs

```diff
@@ -136,14 +139,13 @@ impl DaVerifier for BitcoinVerifier {
                     let wtxids = inclusion_proof
                         .wtxids
                         .iter()
-                        .copied()
-                        .map(Txid::from_byte_array);
+                        .map(|wtxid| Txid::from_slice(wtxid).unwrap());
```

Comment:
```
            fn from_slice(sl: &[u8]) -> Result<Hash<$($gen),*>, FromSliceError> {
                if sl.len() != $bits / 8 {
                    Err(FromSliceError{expected: Self::LEN, got: sl.len()})
                } else {
                    let mut ret = [0; $bits / 8];
                    ret.copy_from_slice(sl);
                    Ok(Self::internal_new(ret))
                }
            }
                        fn from_byte_array(bytes: Self::Bytes) -> Self {
                Self::internal_new(bytes)
            }
```

So you added an extra .unwrap

Reply:
But if we don't use from_slice, we have to copy the array as it was before. Even if we do not use `copied()` and use it as `from_byte_array(*wtxid)`, but this will again copy the data.

Reply:
fixed